### DISCUSSION TITLE
fajita and mulch webview

### DISCUSSION THREAD

**Discussion by ghost** [Divested-Mobile/DivestOS-Build#71] [Q&A]
Hello,

I made the plunge from LOS to Divest a few weeks ago and so far everything seems to be working well apart from (I think) Webview.

I'm on fajita (oneplus 6t) and I use DeepL translator app by jamal from FDroid. The app uses webview, to make it look like a native app which used to work fine on LOS. On divestOS however, the app just opens a new tab in Mull. And if I switch back to DeepL app using the multitasking view, there is just black screen. I went from march 1st to march 18th build and its the same.

 I cannot figure out how to troubleshoot this. Any ideas?

**Comment by SkewedZeppelin** [comment]
Indeed, I can reproduce this.
That is very strange.
I do not see any errors in logcat either.

Also Mulch WebView updates independently of system build via F-Droid.

**Reply by ghost** [reply]
The reason why I thought webview updates with Divest is because I don't recall it updating in the few weeks I have been using. Before I updated to march 18th build, FDroid was offering me 'CalyxOS Android Webview' update which I found weird but thought nothing of it at the time. 
I did think to check 'installed apps' in FDroid just now and there it is again, 'CalyxOS Android Webview'. Mulch webview is not mentioned as installed nor does using search in FDroid yield any results. It only finds Mulch the browser.

Checking under developer options however, it correctly shows Mulch System Webview.

Could this be somehow connected to it seemingly not working?

**Comment by SkewedZeppelin** [comment]
F-Droid groups packages together with the same ID.
So if you have multiple repos with the system WebView they will share the same
app page.
But if you scroll down to the versions it offers and expand each you'll see some
are offered by the DivestOS repo.

If you don't see that then verify you have the DivestOS WebView still enabled.
The WebView updates are out of the build system for faster security updates:
https://divestos.org/misc/ch-dates.txt

This is a bit messy and I do plan to switch Mulch WebView to its own package ID
soonish.

On the very latest DivestOS release you can now optionally use the Bromite
WebView, perhaps try and see if this DeepL app is still an issue with it.
If it can be reproduced on both Mulch and Bromite it is best to report to the
app developer.

**Reply by ghost** [reply]
It is indeed listed as being from DivestOS repo. I went ahead and updated to the latest build, installed bromite webview, enabled it as default and DeepL does work as it should. Appreciate the option to jump between webviews with such ease and without gymnastics!

Thank you!
